Sources/Sentry/SentryProfiler.mm

```diff
@@ -64,6 +64,85 @@
     return [symbolNSStr substringWithRange:[match rangeAtIndex:1]];
 }
 
+void
+processBacktrace(const Backtrace &backtrace,
+    NSMutableDictionary<NSString *, NSMutableDictionary *> *threadMetadata,
+    NSMutableDictionary<NSString *, NSDictionary *> *queueMetadata,
+    NSMutableArray<NSDictionary<NSString *, id> *> *samples,
+    NSMutableArray<NSMutableArray<NSNumber *> *> *stacks,
+    NSMutableArray<NSDictionary<NSString *, id> *> *frames,
+    NSMutableDictionary<NSString *, NSNumber *> *frameIndexLookup, uint64_t startTimestamp,
+    NSMutableDictionary<NSArray<NSNumber *> *, NSNumber *> *stackIndexLookup)
```

Comment:
do we know if using an `NSArray` as a dictionary key works as expected? I'm guessing it computes a reasonable hash but just wondering

Reply:
We're using an `NSArray<NSNumber *>`, and both of those classes conform to NSCoding and NSCopying which are the requirements for NSDictionary keys. The tests that passed with the slow comparison still pass with the hash approach.

But now that you mention it, copying the whole array to use as a key is going to be slower and less memory efficient than just computing the hash and using that as the key. I will make that change.

Reply:
OK, wound up going the route of constructing string keys from the stacks since NSArray.hash only returns the array length ಠ_ಠ